src/main/native/compression/IntelDeflater.cc

```diff
@@ -90,8 +90,11 @@ JNIEXPORT void JNICALL Java_com_intel_gkl_compression_IntelDeflater_resetNative
     if (lz_stream == 0) {
       lz_stream = (isal_zstream*)malloc(sizeof(isal_zstream));
       if ( lz_stream == NULL ) {
-        jclass Exception = env->FindClass("java/lang/Exception");
-        env->ThrowNew(Exception,"Memory allocation error");
+
+        if(env->ExceptionCheck())
+            env->ExceptionClear();
+
```

Comment:
Do we expect any exception here?

Reply:
As discussed during design/code reviews, we might expect exceptions from malloc and calloc operations like L#91 so this exception check is to handle that.